### ISSUE TITLE
Quoted coma in a cell 

### ISSUE THREAD

**Issue by lessless** [beatrichartz/csv#81]
Hi, 
I want to discuss another edge case in which we bumped last Friday. User provided us a CSV that looks like this 
```
test30000@example.com,First30000,Last30000
test30001@example.com,First30001,Last30001
test30003@example.com,First","30001,Last30001
```
As you see second cell of the last row has quoted delimiter character and  here is how it is decoded. 

```
iex(6)> File.stream!(file) |> CSV.decode!() |> Enum.into([])
[
  ["test30000@example.com", "First30000", "Last30000"],
  ["test30001@example.com", "First30001", "Last30001"],
  ["test30003@example.com", "First,30001", "Last30001"]
]
```
That plays badly with our importing routine - database thinks that last line actually has 4 columns.

While fixing this is not a problem I would like to know that unquoting delimiting character is an intended behavior and we can rely on fact that it won't change in the future

**Comment by beatrichartz** [comment]
Thanks for raising this. Looks like a bug to me. I'd expect this to decode to:
```elixir
[
  ["test30000@example.com", "First30000", "Last30000"],
  ["test30001@example.com", "First30001", "Last30001"],
  ["test30003@example.com", "First\"", "'\"30001", "Last30001"]
]
```
Is this what you'd expect as well?

**Comment by beatrichartz** [comment]
Although on second look the input is missing a preceding quote to escape the quotes in the field - i.e. correctly formed input would be:
```
test30003@example.com,First"",""30001,Last30001
```

**Comment by lessless** [comment]
How this will look like in Elixir terms?

**Comment by beatrichartz** [comment]
Well-formed input would look like:

`test30003@example.com,First"",""30001,Last30001`

The output for this currently looks like this:

```elixir
File.stream!("file.csv") |> CSV.decode!() |> Enum.into([])
[
  ["test30000@example.com", "First30000", "Last30000"],
  ["test30001@example.com", "First30001", "Last30001"],
  ["test30003@example.com", "First", "30001", "Last30001"]
]
````

According to spec, it should look like this:

```elixir
[
  ["test30000@example.com", "First30000", "Last30000"],
  ["test30001@example.com", "First30001", "Last30001"],
  ["test30003@example.com", "First\"", "\"30001", "Last30001"]
]
```

**Comment by lessless** [comment]
Ah, got you know. Well, users will come up with all kinds of sequences in their CSVs :) 
I do not know how a field with a single quote came to life - maybe somebody tried to make fun when they entered their credentials in subscription field, but here we are.

As I told - everything will work, just would love to know that it's a deliberate decision and it will not change in the future. 
Is there anything about single quotes in the spec?

**Comment by beatrichartz** [comment]
Interesting - what is writing the CSV you're parsing in?

The idea is to follow the spec as closely as possible, so the input you presented should be an erroneous row, and escaped quotes should be included in the output, as in the comment above.

**Comment by lessless** [comment]
Honestly, don't know. If that line is broken according to the spec, then I don't have any further input here :)

**Comment by beatrichartz** [comment]
This is fixed and released in 2.3.0 - just to add a realisation from my side regarding this line: 

```
test30003@example.com,First","30001,Last30001
```
I think this is trying to escape the comma by enclosing the comma in double quotes, however, the spec would need commas to be escaped by enclosing the whole field in double quotes, e.g.

```
test30003@example.com,"First,30001",Last30001
```

Leading to this output:

```elixir
["test30003@example.com", "First,30001", "Last30001"]
```

The other input that will be accepted from 2.3.0 on as above:

```
test30003@example.com,First"",""30001,Last30001
```

which will lead to this output:

```elixir
["test30003@example.com", "First\"", "\"30001", "Last30001"]
```
